src/Components/__tests__/AddressForm.jest.tsx

```diff
@@ -0,0 +1,42 @@
+import { screen, render, fireEvent } from "@testing-library/react"
+import { AddressForm } from "Components/AddressForm"
+import { FC, useState } from "react"
+
+describe("AddressForm", () => {
+  it("preserves updated value from props after user input", () => {
+    const Page: FC = () => {
+      const [address, setAddress] = useState({ addressLine1: "Before update" })
+      const updateAddress = () => {
+        setAddress({ addressLine1: "After update" })
+      }
+
+      return (
+        <>
+          <AddressForm
+            value={address}
+            errors={{}}
+            touched={{}}
+            onChange={address => {
+              setAddress(address)
+            }}
+          />
+          <button onClick={updateAddress}>Update</button>
+        </>
+      )
+    }
+
+    render(<Page />)
+
+    const line1 = screen.getByPlaceholderText(
+      "Street address"
+    ) as HTMLInputElement
+    expect(line1.value).toBe("Before update")
+
+    fireEvent.click(screen.getByText("Update"))
+    expect(line1.value).toBe("After update")
+
+    const city = screen.getByPlaceholderText("City")
+    fireEvent.change(city, { target: { value: "New York" } })
+    expect(line1.value).toBe("After update")
```

Comment:
This test [reproduces the issue](https://app.circleci.com/pipelines/github/artsy/force/51173/workflows/c20d9ce9-3454-4a80-8007-6bf27c743441/jobs/383361).

![Screen Shot 2023-08-11 at 10 05 49 PM](https://github.com/artsy/force/assets/796573/fbdea649-ac25-48ed-8e0d-93a9359e27a3)